### DISCUSSION TITLE
Using Microsoft.Extensions.Http.Resilience

### DISCUSSION THREAD

**Discussion by danielwertheim** [dotnet/yarp#2502] [General] [answered]
Can the Microsoft.Extensions.Http.Resilience package be used with YARP or do YARP have it's own HttpClient-factory etc meaning e.g `AddStandardResilienceHandler()` as shown here: https://learn.microsoft.com/en-us/dotnet/core/resilience/http-resilience?tabs=dotnet-cli will not work?

**Comment by MihaZupan** [comment, marked as answer]
YARP does not use `IHttpClientFactory`, but a custom [`IForwarderHttpClientFactory`](https://github.com/microsoft/reverse-proxy/blob/main/src/ReverseProxy/Forwarder/IForwarderHttpClientFactory.cs), so you can't use the factory extensions.

You're able to inject your custom handlers into the `HttpClient`, but retires at that layer may not be the most effective with YARP.
That is, instead of retrying a request to the same destination, you likely want to try a different one and let YARP know that the first one is unhealthy.

There's some more discussion about this in #56.
